tests/test_decision_maker/test_base.py

```diff
@@ -138,6 +139,7 @@ def test_transaction_is_affordable(self):
         assert not self.ownership_state.check_transaction_is_affordable(tx_message=tx_message), \
             "We should reject the transaction!"
 
+        # Create a message that we expect to be rejected because we don't have enough tokens.
```

Comment:
How are the two tests different, this and the before? what am I missing?

Reply:
Well these 2 tests are different because they cover different part of the code. If you see following code : 
`        if tx_message.amount == 0 and all(quantity == 0 for quantity in tx_message.tx_quantities_by_good_id.values()):
            # reject the transaction when there is no wealth exchange
            result = False
        elif tx_message.amount <= 0 and all(quantity >= 0 for quantity in tx_message.tx_quantities_by_good_id.values()):
            # check if the agent has the money to cover the sender_amount (the agent=sender is the buyer)
            result = self.amount_by_currency_id[tx_message.currency_id] >= tx_message.sender_amount
        elif tx_message.amount >= 0 and all(quantity <= 0 for quantity in tx_message.tx_quantities_by_good_id.values()):
            # check if the agent has the goods (the agent=sender is the seller).
            result = all(self.quantities_by_good_id[good_id] >= -quantity for good_id, quantity in tx_message.tx_quantities_by_good_id.items())
        else:
            result = False
        return result`
You will see that by providing in the first test 0 as values for quantity and amount will satisfy a different if statement.